light-client/examples/light_client.rs

```diff
@@ -109,27 +110,31 @@ fn sync_cmd(opts: SyncOpts) {
     let verifier = ProdVerifier::default();
     let clock = SystemClock;
     let scheduler = scheduler::basic_bisecting_schedule;
-    let fork_detector = ProdForkDetector::default();
-
-    let mut light_client = LightClient::new(
-        state,
-        options,
-        clock,
-        scheduler,
-        verifier,
-        fork_detector,
-        io,
-    );
+
+    let light_client = LightClient::new(primary, options, clock, scheduler, verifier, io);
+
+    let instance = Instance::new(light_client, state);
+
+    let peer_list = PeerList::builder()
+        .primary(primary)
+        .peer(primary, instance)
+        .build();
+
+    let mut supervisor = Supervisor::new(peer_list, ProdForkDetector::default());
+    let mut handle = supervisor.handle();
+
+    std::thread::spawn(|| supervisor.run());
 
     loop {
-        match light_client.verify_to_highest() {
+        handle.verify_to_highest_async(|result| match result {
             Ok(light_block) => {
                 println!("[ info  ] synced to block {}", light_block.height());
             }
             Err(e) => {
                 println!("[ error ] sync failed: {}", e);
             }
-        }
+        });
+
         std::thread::sleep(Duration::from_millis(800));
```

Comment:
fyi: in Go, we don't have a runtime because we don't know what is a single best runtime. for some, a loop like ^ might be sufficient (e.g. check for a new block every 5s). others may want to subscribe for new headers via WebSocket and verify headers as soon as there are committed by Tendermint.

Reply:
although, this is an example, so feel free to ignore